java/src/jni_client.rs

```diff
@@ -10,13 +10,37 @@ use jni::JNIEnv;
 use jni::JavaVM;
 use jni::objects::{GlobalRef, JClass, JObject, JStaticMethodID, JValue};
 use jni::signature;
-use jni::sys::{jlong, jstring};
+use jni::sys::{jint, jlong, jstring};
 use redis::{RedisError, Value as ServerValue};
+use std::ffi::c_void;
 use std::sync::Arc;
 use std::sync::mpsc::{Sender, channel};
 use std::thread;
 use tokio::runtime::Runtime;
 
+const JNI_VERSION_1_8: jint = 0x00010008;
+
+#[unsafe(no_mangle)]
+pub extern "system" fn JNI_OnLoad(vm: JavaVM, _reserved: *mut c_void) -> jint {
+    // Cache the JavaVM for later use
+    let _ = JVM.set(Arc::new(vm));
+
+    // Pre-cache only MethodCache with correct classloader context
+    // GlideCoreClientCache will be cached later
+    if let Some(jvm) = JVM.get()
+        && let Ok(mut env) = jvm.get_env()
+    {
+        let _ = get_method_cache(&mut env);
+    }
+
+    JNI_VERSION_1_8
+}
+
+#[unsafe(no_mangle)]
+pub extern "system" fn JNI_OnUnload(_vm: JavaVM, _reserved: *mut c_void) {
+    // Clean up global references to prevent memory leaks
```

Comment:
We need to implement this to explicit cleanup the static cache objects. Rust doesn't invoke Drop on statics, so the GlobalRef objects won't get Drop'd and subsequently won't be released by the JVM.

This can happen if the calling code repeatedly loads and unloads the library (which we used to do on every connection, but this got fixed).

Reference: https://internals.rust-lang.org/t/when-a-dll-built-with-rust-is-unloaded-it-seems-that-rust-does-not-automatically-call-drop-on-static-global-variables-defined-in-the-dll/23153/2